src/lib/vault/encryptFormSubmission.ts

```diff
@@ -0,0 +1,40 @@
+import { getPublicKey } from "./getPublicKey.js";
+import type { FormSubmission } from "./dataStructures/formSubmission.js";
+import {
+  randomBytes,
+  createCipheriv,
+  createPublicKey,
+  publicEncrypt,
+} from "node:crypto";
+
+export const encryptFormSubmission = async (
+  serviceAccountId: string,
+  submission: FormSubmission,
+) => {
+  const serviceAccountPublicKey = await getPublicKey(serviceAccountId);
+
+  // Encrypt the submission with the public key
+  const encryptionKey = randomBytes(32);
+  const iv = randomBytes(16);
+
+  const cipher = createCipheriv("aes-256-gcm", encryptionKey, iv);
+
+  const encryptedResponses = Buffer.concat([
+    cipher.update(Buffer.from(JSON.stringify(submission))),
+    cipher.final(),
+  ]);
+  const authTag = cipher.getAuthTag();
+  const publicKey = createPublicKey({ key: serviceAccountPublicKey });
+  const encryptedKey = publicEncrypt(publicKey, encryptionKey).toString(
+    "base64",
+  );
+  const encryptedNonce = publicEncrypt(publicKey, iv).toString("base64");
+  const encryptedAuthTag = publicEncrypt(publicKey, authTag).toString("base64");
+
+  return {
+    encryptedResponses,
+    encryptedKey,
+    encryptedNonce,
+    encryptedAuthTag,
+  };
```

Comment:
For a future refactoring I feel like this function could be split into two separated ones. 
- One would be for just encrypting a string with a public key (2 inputs). 
- The other would be for assembling everything together (getPublicKey then encrypt string then return result). It would become the function used by the API endpoint.

Reply:
I think I understand where you're coming from.  More then happy to discuss when refactoring in the future.